caas/kubernetes/provider/pebble/pebble.go

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package pebble
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// Probe constants
+const (
+	alivePath = "/v1/health?level=alive"
+	readyPath = "/v1/health?level=ready"
+)
+
+func LivenessHandler(port string) corev1.ProbeHandler {
+	return corev1.ProbeHandler{
+		HTTPGet: &corev1.HTTPGetAction{
+			Path: alivePath,
+			Port: intstr.Parse(port),
+		},
+	}
+}
+
+func ReadinessHandler(port string) corev1.ProbeHandler {
+	return corev1.ProbeHandler{
+		HTTPGet: &corev1.HTTPGetAction{
+			Path: readyPath,
+			Port: intstr.Parse(port),
+		},
+	}
+}
+
+// Reserved Pebble health check ports for each container.
+// All ports from 38813 onwards are reserved for workload containers
+// Additional ports should be allocated backwards
+// e.g. 38812, 38811, 38810, ...
+const (
+	// for api-server container (jujud)
+	ApiServerHealthCheckPort = "38811"
+
+	// for charm container (containeragent)
+	CharmHealthCheckPort = "38812"
+
+	// Arbitrary, but PEBBLE -> P38813 -> Port 38813
+	// workload containers start at this value and increment
+	// see WorkloadHealthCheckPort function below
+	workloadHealthCheckPortStart = 38813
+)
+
+// Returns Pebble port for workload container i
+func WorkloadHealthCheckPort(i int) string {
```

Comment:
It might be worth creating a quicker helper class that you can just call inc on to keep getting a new port number.

The reasoning is that in this scenario it's possible to provide a negative port value that could conflict with other ports.

By having a quicker helper we can just go inc() and read() and it becomes a lot safer.

Not a huge deal just a nit if we are going to make helpers.

Reply:
I'd want to be able to retrieve the port numbers later if needed. I think we can just all agree not to use negative values here